### PR TITLE
[SPARK-50428][SS][PYTHON] Support TransformWithStateInPandas in batch queries

### PR DESCRIPTION
**Pull request by bogao007**
### What changes were proposed in this pull request?

Support TransformWithStateInPandas in batch queries.

### Why are the changes needed?

Bring parity as Scala. Scala batch support for TransformWithState is done in https://issues.apache.org/jira/browse/SPARK-46865.

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

Added new unit test cases.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala [hunk 1]
```diff
@@ -967,18 +968,16 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           keyEncoder, outputObjAttr, planLater(child), hasInitialState,
           initialStateGroupingAttrs, initialStateDataAttrs,
           initialStateDeserializer, planLater(initialState)) :: Nil
+      case t @ TransformWithStateInPandas(
+      func, _, outputAttrs, outputMode, timeMode, child,
```
Comment (anishshri-db) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala:972]: nit: indent seems off ?

**Review by anishshri-db — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala (hunk 1 above)
Comment (bogao007) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: Thanks, fixed.

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**